### PR TITLE
Extend ESLint exercise 5 with the re-assignation case

### LINKED ISSUE #9: Extend ex.es.5's final solution
Hello ✋🏻, I'm trying to make a PR to present to you an extended solution on the fifth ESLint exercise, but I can't seem to understand how exactly ```split-guide``` works. I've created the (with the ```split-guide``` markup) exercise file and its test file inside the ```templates``` directory (which is on the root of the project) and I've issued ```yarn start split``` but nothing happens, nothing updates on the file system, even though I get this output:

```javascript
yarn start v0.27.5
$ nps "split"
nps is executing `split` : node node_modules/concurrently/src/main.js --kill-others-on-fail --prefix-colors "bgBlue.bold,bgMagenta.bold" --prefix "[{name}]" --names "split.exercises,split.demos" 'nps split.exercises' 'nps split.demos'
[split.exercises] nps is executing `split.exercises` : split-guide generate --templates-dir other/old-exercises/templates --exercises-dir other/old-exercises/exercises --exercises-final-dir other/old-exercises/exercises-final --silent-success
[split.demos] nps is executing `split.demos` : split-guide generate --templates-dir other/old-demos/templates --exercises-dir other/old-demos/start --exercises-final-dir other/old-demos/final --silent-success
[split.demos] nps split.demos exited with code 0
[split.exercises] nps split.exercises exited with code 0
Done in 2.31s.
```
What am I doing wrong?

### PR DESCRIPTION
**Pull request by mstaicu**
I've extended the fifth exercise from the ESLint series in order to cover re-assignations of the variable that was originally been assigned the 'console' object. If such a behavior is not desired, let me know. Fixes #9

### CODE REVIEW

**Comment by mstaicu** [comment]
Not sure why ESLint is telling me that I should return something from the arrow function.

**Comment by kentcdodds** [comment]
If you want, you can run the `validate` script before you commit to make sure that nothing's wrong before you push things up. Also, since this is your first PR, you might consider watching [this](http://kcd.im/pull-request). It's been pretty helpful to a lot of folks :smile:

File: other/final/no-console-5.js [hunk 1]
```diff
@@ -85,6 +51,53 @@ module.exports = {
...
+            .references.slice(1)
+
+          references.forEach(reference => {
+            if (
+              findParent(
+                reference.identifier,
+                parent => parent.type === 'VariableDeclarator',
+              )
+            ) {
+              return reportDeniedMethodUsage(reference.identifier)
```
Comment (kentcdodds) [inline other/final/no-console-5.js]: I have a rule enabled called `consistent-return` which requires that if you return something anywhere you have to return something everywhere. Makes things more consistent. In this case we don't actually care about the return value, but the code makes it look like the return value does matter. So to fix this, just put the return statement on the next line by itself :smile:

**Comment by mstaicu** [comment]
Your course on OSS on FEM was really awesome. It was like a HIIT for programming skills, it worked a lot of concepts in a pragmatic way. And yes, in the midst excitement, I forgot to validate or even check if there are validation scripts, that's what I loved about ```husky``` or ```git-hooks```, set it once, never worry again. I'm doing another commit with the passing validations 💪🏻. We can talk more about your idea for exercise 6 after this.

**Comment by mstaicu** [comment]
Now that you've seen the code, and you proposed to create a new exercise, should we close this issue and I should go back and create the new exercise, on a new branch, and then come back and do a fresh PR? If yes, then I have the following question. I assume the exercise should be in 'chronological' order, thus meaning that it should indeed come after exercise 5 of the ESLint exercise series. But currently there's exercise 6 as the last one, so I was thinking that my proposal should be exercise 6, as you stated as well, and the current exercise 6 should become 7, or should I just merge it all into the existing exercise 6? If I issue ```npm start ex.es.6``` I don't get any tests, so I could create the ```no-console-7.test.js``` file with the corresponding test suite.

That's .. a lot to read, I should put in a tl;dr; 😄, I hope you can make it to the end!

**Comment by kentcdodds** [comment]
Ah, yeah, what don't we make this the very last exercise? That way it doesn't disrupt the current for of exercises?

**Comment by mstaicu** [comment]
Sure thing, I'll whip up a new branch with a new commit.

**Comment by mstaicu** [comment]
I think we can close this as well.

**Pull request closed without merging**